core/src/main/java/com/datastrato/gravitino/metrics/MetricsSystem.java

```diff
@@ -31,19 +32,22 @@
  * report metrics from MetricsSources registered to MetricsSystem.
  */
 public class MetricsSystem implements Closeable {
+
   private static final Logger LOG = LoggerFactory.getLogger(MetricsSystem.class);
   private final String name;
   private final MetricRegistry metricRegistry;
   private HashMap<String, MetricsSource> metricSources = new HashMap<>();
   private List<Reporter> metricsReporters = new ArrayList<>();
   private CollectorRegistry prometheusRegistry;
+  private Config serverConfig;
 
-  public MetricsSystem() {
-    this("");
+  public MetricsSystem(Config config) {
```

Comment:
Now, you directly pass the `Config` to MetricsSource, this is not needed, why do you still need to change this?

Reply:
We couldn't get `ServerConfig` in `IcebergRESTService`, so saving `Config` to `MetricsSystem` to get it in `IcebergRESTService`

Reply:
It's a bit hacky, can you please figure out a better way. Also why don't you pass in as a Iceberg rest service config, so that it can be used here.

Reply:
You mean retrieve the metrics config from `ServerConfig` and pass in as `Iceberg REST service` config?  this also a little hacky

Reply:
I'm you can pass the aux configuration "aux.xxx" to iceberg rest service, so the rest service can get and use it, do you think it is hacky?

Reply:
the properties passed to Iceberg REST service are properties with `gravitino.auxService.iceberg-rest.` prefix, I think it's hacky to pass `gravitino.metrics.timeSlidingWindowSecs` to the properties.  Maybe we could pass extra `Config` object to aux service to pass generic configurations.